src/libsyntax/std_inject.rs

```diff
@@ -58,14 +66,27 @@ pub fn maybe_inject_crates_ref(mut krate: ast::Crate, alt_std_name: Option<&str>
     };
 
     // .rev() to preserve ordering above in combination with insert(0, ...)
-    for name in names.iter().rev() {
+    let alt_std_name = alt_std_name.map(Symbol::intern);
+    for orig_name in names.iter().rev() {
+        let orig_name = Symbol::intern(orig_name);
+        let mut rename = orig_name;
+        // HACK(eddyb) gensym the injected crates on the Rust 2018 edition,
+        // so they don't accidentally interfere with the new import paths.
```

Comment:
This is not even a hack, if the new module system is adopted we need to actively avoid placing an item named `std` (or `core`) into the root module, so that stuff like `super::std` or `crate::std` untypical for the new model doesn't resolve and doesn't break uniformity.

Reply:
Yes but I'm not sure what the best approach is. I'd say "remove them completely" but then you probably need a way to link against them even if no item from them is *actually* used.